[DM-2145] [OPCUA] Change log for device type validation and alarm creation on device type matching failure (#2367)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-10-20-79-0-added-validation-on-device-type.md

---

> date:

Reviewer: ```suggestion
date: 2024-09-02
```

> title: Added validation for device type resources API and alarm creation on device type matching failure

Reviewer 2: "device type matching failure" is difficult to read and understand. I would just say "Added validation and alarming for device type resources API" For what exactly an alarm will be triggered will be explained in the details. I think that's sufficient.

> title: Added validation for device type resources API and alarm creation on device type matching failure

Reviewer 3: ```suggestion
title: Added validation and alarming for device type resources API
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-2145
version: 10.20.79.0
---

> A new validation for the device type API (CREATE and UPDATE operations) has been implemented to prevent duplicate 'browsePath' entries in the mappings. 

Reviewer 3: ```suggestion
A new validation for the device type API (CREATE and UPDATE operations) has been implemented to prevent duplicate ´browsePath´ entries in the mappings. 
```

> A new validation for the device type API (CREATE and UPDATE operations) has been implemented to prevent duplicate 'browsePath' entries in the mappings. 

Reviewer 3: @sourajitsaha17 What is the device type API? I can´t find it in the API specification. Shouldn´t it be described there?

Author: @BeateRixen it is documented here https://cumulocity.com/docs/protocol-integration/opcua/#device-type-resources Maybe I should add 'resources' also?

Reviewer 3: @sourajitsaha17 I was not aware that this is an OPCUA specific API. I think device type is sufficient then. Is the change documented? A link to the documentation could help, for example:
[device type API](/protocol-integration/opcua/#device-type-resources)

Author: actually, I didn't think about that. Can I add the documentation in the same PR or should it be separate?

Reviewer 3: If you put in in a separate PR you could already merge this change log PR as it will not go live without a date. If you add it here, you can only merge the entire PR when the feature is deployed.

Author: Thanks for the clarification :) I added to this one and labeled as 'not-yet-deployed'.

> A new validation for the device type resources API of OPC UA service (CREATE and UPDATE operations) has been implemented to prevent duplicate `browsePath` entries in the mappings. Refer to [device type API](/protocol-integration/opcua/rest-api) for details about browse path. 

Reviewer 3: ```suggestion
A new validation for the device type resources API of the OPC UA service (CREATE and UPDATE operations) has been implemented to prevent duplicate `browsePath` entries in the mappings. Refer to [Device type resources](protocol-integration/opcua/#device-type-resources) for details about the browse path. 
```
